PWGCF/MultiparticleCorrelations/Tasks/multiparticle-correlations-ar.cxx

```diff
@@ -385,33 +610,505 @@ struct MultiParticleCorrelationsARTask {
            AR::SurviveCut(cfgITSCLUSTERS.value, track.itsNCls());
   }
 
+  template <typename TrackObject>
+  void FillAzimuthalAngle(TrackObject track)
+  {
+    double angle = track.phi();
+    double weight = GetWeight<TrackObject>(track);
+
+    // all event angles, for computing correlators as a function of event variable
+    fAzimuthalAnglesAll.push_back(angle);
+    fWeightsAll.push_back(weight);
+
+    // fill angles into bins of a track variable, for computing correlators as a function of track variable
+    std::array<double, AR::kLAST_CorTrackDep> TrackDep = {track.pt(), track.eta()};
+    int bin;
+    for (int track = 0; track < AR::kLAST_CorTrackDep; track++) {
+      bin = fTrackDepHists[track]->FindBin(TrackDep[track]) - 1; // in root, bins start at 1
+      fAzimuthalAnglesTrackDep[track].at(bin).push_back(angle);
+      fWeightsTrackDep[track].at(bin).push_back(weight);
+    }
```

Comment:
Do you need to do this loop on a per track basis?

Reply:
I store a copy of the angle for each bin seperately (like I mentioned above).

When I first came up with this algorithm, instead of storing the value of angle, I stored the index of where that angle is in vector holding all the angles in the event. But whether I store the index (an integer) or the value (a double) should not make a difference.